icicle/backend/cpu/include/cpu_program_executor.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <vector>
+#include <functional>
+#include "icicle/program/symbol.h"
+#include "icicle/program/program.h"
+
+namespace icicle {
+
+  template <typename S>
+  class CpuProgramExecutor
+  {
+  public:
+    CpuProgramExecutor(Program<S>& program)
```

Comment:
I think you need to either take is as const reference or by value. Why would you modify the input program and make it not work on other backends possible?

Reply:
done